### ISSUE TITLE
Gem remote not being created correctly

### ISSUE THREAD

**Issue by mcaulifn** [jfrog/terraform-provider-artifactory#635]
**Describe the bug**
When creating a gem remote via Terraform, something is not initialized correctly. 

![image](https://user-images.githubusercontent.com/16712138/214686126-8e6b626f-d8d0-4d8f-971f-48dc0ebb5145.png)

**Requirements for and issue**
- [ ] A description of the bug
- [ ] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue). **If this is not supplied, this issue will likely be closed without any effort expended.**
- [ ] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [ ] Your version of terraform
- [ ] Your version of terraform provider

**Expected behavior**
The remote is created and initialized properly.

**Additional context**

```
Terraform v1.3.6
on linux_amd64
+ provider registry.terraform.io/jfrog/artifactory v6.24.1
+ provider registry.terraform.io/jfrog/project v1.1.13
```

```hcl
resource "artifactory_remote_gems_repository" "this" {
  count                                 = var.type == "gems" ? 1 : 0
  key                                   = var.name
  list_remote_folder_items              = true
  password                              = var.password
  propagate_query_params                = true
  store_artifacts_locally               = true
  unused_artifacts_cleanup_period_hours = var.artifact_cleanup_period
  url                                   = var.url
  username                              = var.username

  content_synchronisation {
    enabled                         = var.content_synchronisation_enabled
    properties_enabled              = var.content_synchronisation_enabled
    source_origin_absence_detection = var.content_synchronisation_enabled
    statistics_enabled              = var.content_synchronisation_enabled
  }
}
```

**Comment by danielmkn** [comment]
Hi @mcaulifn, can you please provide a bit more info on what's wrong with the repo?

**Comment by mcaulifn** [comment]
I am getting this when attempting to add it as a source. 

```
$ gem source -a https://user:pass@host/artifactory/api/gems/public-gems-virtual
Error fetching https://user:pass@host/artifactory/api/gems/public-gems-virtual:
        bad response Not Found 404 (https://user:pass@host/artifactory/api/gems/public-gems-virtual/specs.4.8.gz)
```

It works fine when I manually add the remote.

**Comment by danielmkn** [comment]
Ok, thanks! I'll take a look.

**Comment by mcaulifn** [comment]
@danielmkn Any luck?

**Comment by danielmkn** [comment]
Sorry, I've got to work on this ticket only today, I'll update you by the end of the day.

**Comment by danielmkn** [comment]
@mcaulifn can you please provide all the values of the vars you used in the resource? I can't reproduce the error with default values. 
For me `gem source` works for all three repos I've created to test: one with API, one in the UI and one with the provider. 
My provider repo attributes: 
```
resource "artifactory_remote_gems_repository" "gem-remote-provider" {
  key = "gem-remote-provider"
  url = "https://rubygems.org/"
}

Command I've ran: `gem source -a http://admin:PASSWORD@host/artifactory/api/gems/gem-remote-provider/`, note the slash at the end. 
Response: `http://admin:PASSWORD@host/artifactory/api/gems/gem-remote-api/ added to sources`

**Comment by mcaulifn** [comment]
Same thing with the `/` at the end. 

This is a `terraform show` from the platform node.

```
resource "artifactory_remote_gems_repository" "this" {
    allow_any_host_auth                   = false
    assumed_offline_period_secs           = 300
    blacked_out                           = false
    block_mismatching_mime_types          = true
    bypass_head_requests                  = false
    client_tls_certificate                = ""
    description                           = " (local file cache)"
    download_direct                       = false
    enable_cookie_management              = false
    excludes_pattern                      = ""
    hard_fail                             = false
    id                                    = "rubygems-gems"
    includes_pattern                      = "**/*"
    key                                   = "rubygems-gems"
    list_remote_folder_items              = true
    local_address                         = ""
    metadata_retrieval_timeout_secs       = 60
    mismatching_mime_types_override_list  = ""
    missed_cache_period_seconds           = 1800
    notes                                 = ""
    offline                               = false
    package_type                          = "gems"
    priority_resolution                   = false
    project_environments                  = []
    project_key                           = ""
    propagate_query_params                = true
    property_sets                         = []
    proxy                                 = ""
    query_params                          = ""
    remote_repo_layout_ref                = ""
    repo_layout_ref                       = "simple-default"
    retrieval_cache_period_seconds        = 7200
    share_configuration                   = false
    socket_timeout_millis                 = 15000
    store_artifacts_locally               = true
    synchronize_properties                = false
    unused_artifacts_cleanup_period_hours = 720
    url                                   = "https://rubygems.org/"
    username                              = (sensitive value)
    xray_index                            = false

    content_synchronisation {
        enabled                         = false
        properties_enabled              = false
        source_origin_absence_detection = false
        statistics_enabled              = false
    }
}
```

Though my command shows a virtual on an edge node, we did the tests from the platform node remote and the edge node remote. There is an internal ticket which lists this issue number if you need additional information.

**Comment by danielmkn** [comment]
I see that if you set `propagate_query_params` to `false` (which is a default value), `gem sources` works fine. 
That attribute is intended to be used **only with generic repositories**, per [official documentation](https://www.jfrog.com/confluence/display/JFROG/Advanced+Settings).
When the repo is created in the UI or API, this parameter is set to `false`, this is why it works fine. 
Please, set that parameter to `false`, and I'll update the provider documentation. 

Thanks for the ticket, it helps us to improve!

**Comment by mcaulifn** [comment]
Should a smart remote also be set to false? And does this only apply to `gems`?

**Comment by danielmkn** [comment]
Yes, all but Generic should be `false` for Propagate, I'll update the PR, I will restrict this attribute only to be available to Generic repos, and add a schema migrator. 
Content synchronization (smart repo) can be enabled, just make sure you follow the instruction [here](https://www.jfrog.com/confluence/display/JFROG/Smart+Remote+Repositories).

**Comment by danielmkn** [comment]
Fixed in v6.26.0
Updating the provider to this version will remove `propagate_query_params` from the state. Also, this attribute is only available to the Generic repo type.
